up42/catalog.py

```diff
@@ -196,12 +196,31 @@ def search(
             ```
         """
         logger.info(f"Searching catalog with search_parameters: {search_parameters}")
+
+        max_limit = search_parameters["limit"]
+        if max_limit > 500:
+            search_parameters = dict(search_parameters)
+            search_parameters["limit"] = 500
+
         url = f"{self.auth._endpoint()}/catalog/stac/search"
         response_json: dict = self.auth._request("POST", url, search_parameters)
-        logger.info(f"{len(response_json['features'])} results returned.")
-        dst_crs = "EPSG:4326"
-        df = GeoDataFrame.from_features(response_json, crs=dst_crs)
+        features = response_json["features"]
+
+        # Catalog search gives results with >500 items as 50 page pagination tokens.
+        # Pagination ends when both links are the same.
+        while len(features) < max_limit:
+            url_page = response_json["links"][1]["href"]
+            if url_page == response_json["links"][0]["href"]:
+                break
```

Comment:
If this condition is not true in the first loop, it will never be true in the other loops. url_page und response_json never get renewed during the loop.

Reply:
Great catch!!!!

Reply:
Added testcase for pagination exhaustion (catalog_pagination_mock only provides one time 500 results, then only one page with 50 results. Only 550 results are returned even though limit was 614).

Loop should also be more readable now with new variable names.